src/NSwag.Commands/Commands/SwaggerGeneration/TypesToJsonSchemaCommand.cs

```diff
@@ -0,0 +1,153 @@
+            set { Settings.IgnoreObsoleteProperties = value; }
+        }
+
+        [Argument(Name = "AllowReferencesWithProperties", IsRequired = false, Description = "Use $ref references even if additional properties are defined on " +
+            "the object (otherwise allOf/oneOf with $ref is used, default: false).")]
+        public bool AllowReferencesWithProperties
+        {
+            get { return Settings.AllowReferencesWithProperties; }
+            set { Settings.AllowReferencesWithProperties = value; }
+        }
+
+        [Argument(Name = "GenerateKnownTypes", IsRequired = false, Description = "Generate schemas for types in KnownTypeAttribute attributes (default: true).")]
+        public bool GenerateKnownTypes
+        {
+            get { return Settings.GenerateKnownTypes; }
+            set { Settings.GenerateKnownTypes = value; }
+        }
+
+        [Argument(Name = "GenerateXmlObjects", IsRequired = false, Description = "Generate xmlObject representation for definitions (default: false).")]
+        public bool GenerateXmlObjects
+        {
+            get { return Settings.GenerateXmlObjects; }
+            set { Settings.GenerateXmlObjects = value; }
+        }
+
+        [Argument(Name = nameof(AllowAdditionalProperties), IsRequired = false, Description = "If `additionalProperties` on all possible definitions and references are allowed (default: false).")]
+        public bool AllowAdditionalProperties { get; set; }
+
+        protected override async Task<string> RunIsolatedAsync(AssemblyLoader.AssemblyLoader assemblyLoader)
+        {
+            var document = new NJsonSchema.JsonSchema4();
+            var generator = new JsonSchemaGenerator(Settings);
+            var schemaResolver = new JsonSchemaResolver(document, Settings);
+
+#if FullNet
+            var assemblies = PathUtilities.ExpandFileWildcards(AssemblyPaths)
+                .Select(path => Assembly.LoadFrom(path)).ToArray();
+#else
+            var currentDirectory = await DynamicApis.DirectoryGetCurrentDirectoryAsync().ConfigureAwait(false);
+            var assemblies = PathUtilities.ExpandFileWildcards(AssemblyPaths)
+                .Select(path => assemblyLoader.Context.LoadFromAssemblyPath(PathUtilities.MakeAbsolutePath(path, currentDirectory))).ToArray();
+#endif
+
+            var allExportedClassNames = assemblies.SelectMany(a => a.ExportedTypes).Select(t => t.FullName).ToList();
+            var matchedClassNames = ClassNames
+                .SelectMany(n => PathUtilities.FindWildcardMatches(n, allExportedClassNames, '.'))
+                .Distinct();
+
+            foreach (var className in matchedClassNames)
+            {
+                var type = assemblies.Select(a => a.GetType(className)).FirstOrDefault(t => t != null);
+                await generator.GenerateAsync(type, schemaResolver).ConfigureAwait(false);
+            }
+
+            document.AllOf.Add(new JsonSchema4
```

Comment:
Why allOf? You should be able to either define the root type or generate all types into “definitions”

Reply:
> `define the root` 
Does it mean that root definition properties-type from current `allOf` should inlin? I.e. there should be no definition for root object? What if we have 2 roots? `oneOf`? So what should I do? May be example? I use online json validator to check sanity of generated stuff.

> generate all types into “definitions”
Do not get it. I need at least one root type inlined or `allOf` or `oneOf` to get validation. What is example of this?